client.go

```diff
@@ -433,7 +427,7 @@ func (d *Dialer) DialContext(ctx context.Context, urlStr string, requestHeader h
 		return nil, nil, err
 	}
 	netConn = nil // to avoid close in defer.
-	return conn, resp, nil
+	return conn, resp, err
```

Comment:
It's better to use `return conn, resp, nil` because it makes it clear that this is always a successful return from the function.